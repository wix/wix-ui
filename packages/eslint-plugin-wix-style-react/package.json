{
  "name": "eslint-plugin-wix-style-react",
  "version": "1.0.0",
  "description": "Wix Style React Lint Plugin",
  "keywords": [
    "eslint",
    "eslintplugin",
    "eslint-plugin"
  ],
  "author": "yairhaimo",
  "main": "lib/index.js",
  "scripts": {
    "test": "mocha './tests/{,!(fixtures)/**}/*.js' --recursive",
    "release": "wnpm-release --no-shrinkwrap"
  },
  "dependencies": {
    "requireindex": "~1.1.0",
    "resolve": "^1.10.0",
    "semver": "^5.6.0"
  },
  "devDependencies": {
    "assert": "^1.4.1",
    "eslint": "~3.9.1",
    "execa": "^1.0.0",
    "mocha": "^3.1.2",
    "wnpm-ci": "latest"
  },
  "engines": {
    "node": ">=0.10.0"
  },
  "license": "ISC",
  "publishConfig": {
    "registry": "https://registry.npmjs.org/"
  }
}
