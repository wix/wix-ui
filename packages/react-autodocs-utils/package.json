{
  "name": "react-autodocs-utils",
  "version": "3.6.3",
  "description": "Set of utilities for extracting meta information about react components mostly to generate automated documentation",
  "main": "index.js",
  "private": false,
  "scripts": {
    "test": "jest",
    "test:watch": "jest --watch"
  },
  "keywords": [
    "react",
    "documentation",
    "docs",
    "autodocs",
    "generated",
    "automatic",
    "metadata",
    "extract",
    "utils",
    "proptypes"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/wix/react-autodocs-utils.git"
  },
  "publishConfig": {
    "registry": "https://registry.npmjs.org/"
  },
  "bugs": {
    "url": "https://github.com/wix/react-autodocs-utils/issues"
  },
  "author": "Arijus Å ukys <argshook@gmail.com> (http://arijus.net)",
  "license": "MIT",
  "devDependencies": {
    "@types/react": "^16.3.11",
    "cista": "0.0.3",
    "jest": "^24.8.0"
  },
  "dependencies": {
    "@babel/core": "7.11.4",
    "@babel/parser": "7.11.4",
    "@babel/traverse": "^7.5.5",
    "@babel/types": "^7.5.5",
    "doctrine": "^3.0.0",
    "react-docgen": "^4.1.1",
    "react-docgen-typescript": "1.18.0",
    "recast": "0.13.0",
    "typescript": "^2.7.1"
  }
}
