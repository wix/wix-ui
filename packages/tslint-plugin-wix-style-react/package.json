{
  "name": "tslint-plugin-wix-style-react",
  "version": "1.0.0",
  "description": "",
  "main": "tslint.json",
  "scripts": {
    "build": "tsc",
    "test": "jest"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "peerDependencies": {
    "tslint": "^5.0.0",
    "typescript": "^2.4.0"
  },
  "devDependencies": {
    "@types/jest": "^23.0.0",
    "@types/node": "^10.5.2",
    "jest": "^23.1.0",
    "ts-jest": "^22.4.6",
    "typescript": "^2.9.2",
    "tslint": "^5.0.0"
  },
  "jest": {
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js"
    ],
    "transform": {
      "^.+\\.(ts|tsx)$": "<rootDir>/node_modules/ts-jest/preprocessor.js"
    },
    "testMatch": [
      "**/**/*.spec.(ts|tsx)"
    ]
  }
}
