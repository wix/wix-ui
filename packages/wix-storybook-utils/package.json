{
  "name": "wix-storybook-utils",
  "version": "1.0.0",
  "description": "Utilities for automated component documentation within Storybook",
  "author": {
    "name": "Wix",
    "email": "fed-infra@wix.com"
  },
  "license": "MIT",
  "files": [
    "src",
    "dist",
    "*.js"
  ],
  "bugs": {
    "url": "https://github.com/wix/wix-ui/issues"
  },
  "homepage": "https://github.com/wix/wix-ui#readme",
  "main": "./dist/src/index.js",
  "scripts": {
    "pretest": "npm run build && build-storybook",
    "test": "haste test --jest",
    "test:watch": "haste test --jest --watch",
    "posttest": "haste lint",
    "start": "haste start --no-test & npm run storybook",
    "release": "haste release",
    "build": "haste build && import-path --path src",
    "storybook": "start-storybook -p 6006"
  },
  "repository": {
    "type": "git",
    "url": "git+ssh://git@github.com/wix/wix-ui.git"
  },
  "keywords": [
    "storybook",
    "utilities",
    "documentation",
    "generated",
    "automatic",
    "react"
  ],
  "publishConfig": {
    "registry": "https://registry.npmjs.org/"
  },
  "peerDependencies": {
    "react": "^15.4.0"
  },
  "dependencies": {
    "babel-runtime": "^6.25.0",
    "classnames": "^2.2.5",
    "copy-to-clipboard": "^3.0.8",
    "deep-eql": "^3.0.1",
    "github-markdown-css": "^2.9.0",
    "highlight.js": "^9.12.0",
    "import-path": "latest",
    "jsx-to-string": "^1.3.0",
    "loader-utils": "^1.1.0",
    "react-autodocs-utils": "^1.0.0",
    "react-collapse": "^4.0.3",
    "react-docgen": "^2.20.0",
    "react-motion": "^0.5.2",
    "react-remarkable": "^1.1.3",
    "recast": "^0.13.0",
    "wix-style-react": "latest"
  },
  "devDependencies": {
    "@storybook/addon-options": "^3.2.16",
    "@storybook/react": "^3.2.16",
    "enzyme": "^3.0.0",
    "enzyme-adapter-react-15": "^1.0.5",
    "eslint-config-wix": "latest",
    "haste-preset-yoshi": "latest",
    "identity-obj-proxy": "^3.0.0",
    "prop-types": "^15.6.0",
    "react": "^15.6.1",
    "react-dom": "^15.6.1",
    "react-test-renderer": "^15.6.0",
    "wix-ui-test-utils": "^1.0.0"
  },
  "babel": {
    "plugins": [
      "transform-runtime"
    ],
    "presets": [
      "wix"
    ]
  },
  "eslintConfig": {
    "extends": "wix/react",
    "env": {
      "jest": true
    },
    "rules": {
      "react/jsx-handler-names": 0,
      "react/no-find-dom-node": 0,
      "react/no-string-refs": 0,
      "react/jsx-boolean-value": 2,
      "comma-dangle": [
        "error",
        "never"
      ]
    }
  },
  "jest": {
    "setupFiles": [
      "<rootDir>/test/enzyme-setup"
    ],
    "setupTestFrameworkScriptFile": "wix-ui-test-utils/jest-setup",
    "moduleNameMapper": {
      "\\.(css|less|scss)$": "identity-obj-proxy"
    },
    "testPathIgnorePatterns": [
      "<rootDir>/dist/",
      "<rootDir>/node_modules/"
    ]
  },
  "haste": {
    "preset": "yoshi"
  },
  "yoshi": {
    "servers": {
      "cdn": {
        "port": 3201
      }
    },
    "entry": {
      "app": "index.js"
    },
    "externals": {
      "react": "React",
      "react-dom": "ReactDOM"
    }
  }
}
