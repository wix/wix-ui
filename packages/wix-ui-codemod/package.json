{
  "private": false,
  "name": "wix-ui-codemod",
  "version": "1.0.0",
  "license": "MIT",
  "author": {
    "name": "Wix",
    "email": "fed-infra@wix.com"
  },
  "bin": {
    "wix-ui-codemod": "./bin/wix-ui-codemod.js"
  },
  "main": "./dist/src/index.js",
  "files": [
    "bin",
    "dist",
    "*.md"
  ],
  "bugs": {
    "url": "https://github.com/wix/wix-ui/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+ssh://git@github.com/wix/wix-ui.git"
  },
  "publishConfig": {
    "registry": "https://registry.npmjs.org/"
  },
  "scripts": {
    "build": "yoshi build",
    "pretest": "yoshi build",
    "test": "yoshi test",
    "posttest": "yoshi lint",
    "release": "yoshi release"
  },
  "dependencies": {
    "commander": "^4.0.1",
    "jscodeshift": "^0.6.4",
    "jscodeshift-add-imports": "^1.0.3"
  },
  "devDependencies": {
    "@types/jest": "^24.0.0",
    "@types/jscodeshift": "^0.6.3",
    "@types/node": "^12.0.0",
    "ts-jest": "^24.0.2",
    "typescript": "~3.7.0",
    "yoshi": "^4.0.0"
  },
  "jest": {
    "transform": {
      "^.+\\.tsx?$": "ts-jest"
    },
    "testPathIgnorePatterns": [
      "<rootDir>/dist/",
      "<rootDir>/node_modules/"
    ]
  }
}
