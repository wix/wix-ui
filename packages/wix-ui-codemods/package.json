{
  "private": false,
  "name": "wix-ui-codemods",
  "version": "1.0.0",
  "license": "MIT",
  "author": {
    "name": "Wix",
    "email": "fed-infra@wix.com"
  },
  "files": [
    "dist"
  ],
  "publishConfig": {
    "registry": "https://registry.npmjs.org/"
  },
  "scripts": {
    "pretest": "yoshi build",
    "test": "yoshi test",
    "posttest": "yoshi lint",
    "release": "yoshi release"
  },
  "devDependencies": {
    "@types/jest": "^24.0.0",
    "@types/node": "^12.0.0",
    "ts-jest": "^24.0.2",
    "typescript": "~3.7.0",
    "yoshi": "^4.0.0"
  },
  "jest": {
    "transform": {
      "^.+\\.tsx?$": "ts-jest"
    },
    "testPathIgnorePatterns": [
      "<rootDir>/dist/",
      "<rootDir>/node_modules/"
    ]
  },
  "yoshi": {}
}
