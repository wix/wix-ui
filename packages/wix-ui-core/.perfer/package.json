{
  "name": "components-bundles",
  "version": "1.0.0",
  "license": "UNLICENSED",
  "private": true,
  "author": {
    "name": "Mykolas Sindeikis",
    "email": "mykolass@wix.com"
  },
  "scripts": {
    "build": "NODE_ENV=production yoshi build",
    "analyse": "NODE_ENV=production yoshi build --analyze"
  }
}
