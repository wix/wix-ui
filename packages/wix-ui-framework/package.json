{
  "private": false,
  "name": "wix-ui-framework",
  "bin": {
    "wuf": "./bin/wuf.js"
  },
  "version": "2.0.0",
  "author": {
    "name": "Wix",
    "email": "fed-infra@wix.com"
  },
  "files": [
    "dist",
    "src",
    "bin",
    "*.md",
    "*.json"
  ],
  "bugs": {
    "url": "https://github.com/wix/wix-ui/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+ssh://git@github.com/wix/wix-ui.git"
  },
  "scripts": {
    "build": "tsc",
    "pretest": "yoshi lint",
    "test": "jest",
    "test:watch": "jest --watch",
    "lint": "yoshi lint",
    "start": "tsc --watch",
    "wuf": "./bin/wuf.js"
  },
  "license": "MIT",
  "publishConfig": {
    "registry": "https://registry.npmjs.org/"
  },
  "dependencies": {
    "commander": "^2.20.0",
    "deep-diff": "^1.0.2",
    "jscodeshift": "^0.6.4",
    "minimatch": "^3.0.4",
    "ora": "^3.4.0",
    "prompts": "^2.0.4",
    "simple-git": "^1.110.0"
  },
  "jest": {
    "transform": {
      "\\.ts$": "ts-jest"
    },
    "moduleFileExtensions": [
      "ts",
      "js"
    ],
    "testPathIgnorePatterns": [
      "<rootDir>/dist/",
      "<rootDir>/node_modules/",
      "<rootDir>/test-generated/",
      "<rootDir>/src/cli-commands/generate/__test__/templates",
      "<rootDir>/src/cli-commands/generate/__test__/codemods"
    ]
  },
  "devDependencies": {
    "@types/jest": "^24.0.12",
    "@types/node": "^11.13.8",
    "cista": "0.0.3",
    "jest": "^24.7.1",
    "nixt": "^0.5.1",
    "ts-jest": "^24.0.2",
    "tslint-config-yoshi": "^4.1.0",
    "typescript": "^3.4.5",
    "yoshi": "^4.3.0"
  },
  "eslintConfig": {
    "extends": "yoshi",
    "env": {
      "jest": true
    }
  }
}
