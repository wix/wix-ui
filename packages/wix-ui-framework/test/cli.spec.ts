import * as path from 'path';
import * as nixt from 'nixt';

import * as packageJson from '../package.json';

const cli = (...args) => {
  const command = path.resolve(__dirname, '..', 'bin', 'wuf.js');
  return `${command} ${args.join(' ')}`;
};

describe('wuf', () => {
  describe('help', () => {
    it('should be displayed when no params', done => {
      nixt()
        .expect(({ stdout }) => {
          expect(stdout).toMatchSnapshot();
        })
        .run(cli())
        .end(done);
    });

    it('should be displayed when --help', done => {
      nixt()
        .expect(({ stdout }) => {
          expect(stdout).toMatchSnapshot();
        })
        .run(cli('--help'))
        .end(done);
    });
  });

  describe('-v, --version', () => {
    it('should echo version from package.json', done => {
      nixt()
        .expect(({ stdout }) => {
          expect(stdout).toEqual(packageJson.version);
        })
        .run(cli('--version'))
        .end(done);
    });
  });

  describe('generate', () => {
    describe('--help', () => {
      it('should render help text', done => {
        nixt()
          .expect(({ stdout }) => {
            expect(stdout).toMatchSnapshot();
          })
          .run(cli('generate --help'))
          .end(done);
      });
    });

    it('should ask for component name', done => {
      nixt()
        .expect(({ stdout }) => {
          expect(stdout).toMatch(/Component name \(PascalCase\).*Test/);
        })
        .run(cli('generate'))
        .on(/Component name/)
        .respond('Test\n')
        .end(done);
    });
  });

  describe('export-testkits', () => {
    describe('--help', () => {
      it('should render help text', done => {
        nixt()
          .expect(({ stdout }) => {
            expect(stdout).toMatchSnapshot();
          })
          .run(cli('export-testkits --help'))
          .end(done);
      });
    });

    describe('--definitions', () => {
      it('should fail with error, given non existing path', done => {
        nixt()
          .expect(({ stderr }) => {
            expect(stderr).toMatch(
              /ERROR: Definitions file does not exist at "whatever.js"/,
            );
          })
          .run(cli('export-testkits --definitions whatever.js'))
          .end(done);
      });
    });

    it.skip('should ask for component name', done => {
      nixt()
        .expect(({ stdout }) => {
          expect(stdout).toMatch(/Generating.../);
        })
        .run(cli('export-testkits'))
        .end(done);
    });
  });
});
