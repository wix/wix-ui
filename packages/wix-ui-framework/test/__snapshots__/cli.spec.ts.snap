// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wuf export-testkits --help should render help text 1`] = `
"Usage: export-testkits [options]

Generate testkit export file

Options:
  --definitions <string>  Path to testkit definitions. Default is \\".wuf/testkit-definitions.json\\"
  --template <string>     Path to template. Default is \\".wuf/testkits/template.js\\"
  --output <string>       Path to output. Default is \\"testkits/{templatePath.fileName | \\"testkits.js\\"}\\"
  -h, --help              output usage information"
`;

exports[`wuf generate --help should render help text 1`] = `
"Usage: generate [options]

Scaffold file structure from templates

Options:
  --component-name <ComponentName>  Component name
  --description <description>       Component description
  --templates <string>              Path to templates. Default is \\"/generator/templates/\\"
  --codemods <string>               Path to codemods. Default is \\"/generator/codemods/\\"
  -f, --force                       Force component generation in a non clean git repo.
  -h, --help                        output usage information"
`;

exports[`wuf help should be displayed when --help 1`] = `
"Usage: wuf [options] [command]

Options:
  -v, --version              output the version number
  -h, --help                 output usage information

Commands:
  generate [options]         Scaffold file structure from templates
  export-testkits [options]  Generate testkit export file"
`;

exports[`wuf help should be displayed when no params 1`] = `
"Usage: wuf [options] [command]

Options:
  -v, --version              output the version number
  -h, --help                 output usage information

Commands:
  generate [options]         Scaffold file structure from templates
  export-testkits [options]  Generate testkit export file"
`;
